sql/src/main/java/io/crate/metadata/doc/DocIndexMetaData.java

```diff
@@ -625,6 +638,10 @@ public DocIndexMetaData build() {
         return notNullColumns;
     }
 
+    ImmutableList<AnalyzedCheckConstraint> checkConstraints() {
```

Comment:
```suggestion
    List<AnalyzedCheckConstraint> checkConstraints() {
```

I know other properties expose the `ImmutableList`, but leaking implementation details outside class boundaries is usually not a good idea. If you want to change implementation details you also break the API.

Reply:
ok, I usually do, but I had a look around in the code, same class, and followed the pattern

Reply:
ImmutableList<GeneratedReference> generatedColumnReferences() {
        return generatedColumnReferences;
    }

    ImmutableCollection<ColumnIdent> notNullColumns() {
        return notNullColumns;
    }